client/app/screens/CreateMatchScreen.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect } from 'react';
+import { Button, ButtonText, ScrollView, FormControl, FormControlLabel, FormControlLabelText, Input, InputField, View, Text} from "@gluestack-ui/themed";
+import { AppStackParamList } from 'app/navigators/AppNavigator';
+import { NativeStackNavigationProp } from '@react-navigation/native-stack';
+import { colors, typography} from "app/theme"
+
+type CreateMatchScreenNavigationProp = NativeStackNavigationProp<
+  AppStackParamList,
+  'Welcome'
+>
+
+export interface CreateMatchScreenProps {
+  navigation: CreateMatchScreenNavigationProp;
+}
+
+/**
+ * CreateMatchScreen component allows users to create and send new match.
+ * @component
+ */
+export function CreateMatchScreen(props: CreateMatchScreenProps) {
+  const [leagueName, setLeagueName] = React.useState<string>("");
+  const [week, setWeek] = React.useState<string>("");
+  const [teamName, setTeamName] = React.useState<string>("");
+  const [createMatchDisabled, setCreateMatchDisabled] = React.useState<boolean>(true);
+  const NFL_WEEKS = 18;
+
+  // TODO: Allow matches for current/future weeks only. 
+  // Pass the current week as a prop/state to this screen.
+  // Output error message if week is invalid.
+
+  useEffect(() => {
+    if (leagueName !== "" && week !== "" && teamName !== "" && Number(week) <= NFL_WEEKS) {
+      setCreateMatchDisabled(false);
+    }
+    else {
+      setCreateMatchDisabled(true);
+    }
+
+  } , [leagueName, week, teamName]);
+
+  const handleLeagueName = (text: string): void => {
```

Comment:
Because these three methods are one line we can change them to `const handleLeagueName = (text: string): void => setLeagueName(text);`

Reply:
Oh, smart idea, that will look really clean